device/common/include/traccc/clusterization/device/impl/aggregate_cluster.ipp

```diff
@@ -54,10 +54,10 @@ inline void aggregate_cluster(
     scalar totalWeight = 0.;
     point2 mean{0., 0.}, var{0., 0.}, offset{0., 0.};
 
-    const auto module_link = cells[cid + start].module_link;
+    const unsigned int module_idx = cells.module_index().at(cid + start);
```

Comment:
```suggestion
    const unsigned int module_idx = cells.at(cid + start).module_index();
```

Reply:
I really do not see the point. :confused: At that point we're trying to make the life of the compiler harder on purpose. And seeing how oneAPI freaked out about this code, I really don't want to complicate things unnecessarily.

Plus, the whole point of an SoA for me is that we can load a single variable when we just need a single variable. :thinking: